base/linalg/cholesky.jl

```diff
@@ -324,11 +324,13 @@ convert{T}(::Type{CholeskyPivoted{T}},C::CholeskyPivoted) =
     CholeskyPivoted(AbstractMatrix{T}(C.factors),C.uplo,C.piv,C.rank,C.tol,C.info)
 convert{T}(::Type{Factorization{T}}, C::CholeskyPivoted) = convert(CholeskyPivoted{T}, C)
 
-full{T,S}(C::Cholesky{T,S}) = C.uplo == 'U' ? C[:U]'C[:U] : C[:L]*C[:L]'
-function full(F::CholeskyPivoted)
-    ip = invperm(F[:p])
-    return (F[:L] * F[:U])[ip,ip]
-end
+convert(::Type{Matrix}, C::Cholesky) = C.uplo == 'U' ? C[:U]'C[:U] : C[:L]*C[:L]'
```

Comment:
Cheers, we're on the same page then. Are you advocating for additionally making those changes (i.e. attacking #12153 simultaneously) in this PR, rather than in a later PR? Thanks!